common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -167,17 +167,13 @@ public int getDefaultPort() {
     }
   }
 
-  public static void assertValidPort(final int port, TachyonConf tachyonConf) {
-    Preconditions.checkNotNull(tachyonConf);
+  public static void assertValidPort(final int port) {
     Preconditions.checkArgument(port < 65536, "Port must be less than 65536");
-
-    if (!tachyonConf.getBoolean(Constants.IN_TEST_MODE)) {
-      Preconditions.checkArgument(port > 0, "Port is only allowed to be zero in test mode.");
-    }
+    Preconditions.checkArgument(port >= 0, "Port must be non-negative");
```

Comment:
This allows us to have multiple workers on the same machine without their ports colliding. I'll add checks to make sure the master port isn't 0.